tests/Unit/Evolution/DiscontinuousGalerkin/SlopeLimiters/Test_LimiterActions.cpp

```diff
@@ -0,0 +1,361 @@
+  using type = Scalar<DataVector>;
+};
+
+template <size_t Dim>
+struct System {
+  static constexpr const size_t volume_dim = Dim;
+  using variables_tag = Tags::Variables<tmpl::list<Var>>;
+};
+
+class DummyLimiterForTest {
+ public:
+  // Data sent by the limiter to its neighbors
+  struct PackagedData {
+    double mean_;
+  };
+  using package_argument_tags = tmpl::list<Var, Tags::Mesh<2>>;
+  void package_data(const gsl::not_null<PackagedData*> packaged_data,
+                    const Scalar<DataVector>& var, const Mesh<2>& mesh) const
+      noexcept {
+    packaged_data->mean_ = mean_value(get(var), mesh);
+  }
+
+  using limit_tags = tmpl::list<Var>;
+  using limit_argument_tags = tmpl::list<Tags::Mesh<2>, Tags::Element<2>>;
+  void operator()(const gsl::not_null<db::item_type<Var>*> var,
+                  const Mesh<2>& /*mesh*/, const Element<2>& /*element*/,
+                  const std::unordered_map<
+                      std::pair<Direction<2>, ElementId<2>>,
+                      DummyLimiterForTest::PackagedData,
+                      boost::hash<std::pair<Direction<2>, ElementId<2>>>>&
+                      neighbor_packaged_data) const noexcept {
+    // Zero the data as an easy check that the limiter got called
+    get(*var) = 0.;
+    for (const auto& data : neighbor_packaged_data) {
+      get(*var) += data.second.mean_;
+    }
+  }
+
+  void pup(const PUP::er& /*p*/) const noexcept {}
+};
+
+struct LimiterTag {
+  using type = DummyLimiterForTest;
+};
+
+template <size_t Dim, typename Metavariables>
+struct component {
+  using metavariables = Metavariables;
+  using chare_type = ActionTesting::MockArrayChare;
+  using array_index = ElementIndex<Dim>;
+  using const_global_cache_tag_list = tmpl::list<LimiterTag>;
+  using action_list =
+      tmpl::list<SlopeLimiters::Actions::SendData<Metavariables>,
+                 SlopeLimiters::Actions::Limit<Metavariables>>;
+  using simple_tags =
+      db::AddSimpleTags<TemporalId, Tags::Mesh<Dim>, Tags::Element<Dim>,
+                        Tags::ElementMap<Dim>, Var>;
+  using initial_databox = db::compute_databox_type<simple_tags>;
+};
+
+template <size_t Dim>
+struct Metavariables {
+  using component_list = tmpl::list<component<Dim, Metavariables>>;
+  using const_global_cache_tag_list = tmpl::list<>;
+  using limiter = LimiterTag;
+  using system = System<Dim>;
+  using temporal_id = TemporalId;
+  static constexpr bool local_time_stepping = false;
+};
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.Evolution.DG.SlopeLimiters.LimiterActions.Generic",
+                  "[Unit][NumericalAlgorithms][Actions]") {
+  using metavariables = Metavariables<2>;
+  using my_component = component<2, metavariables>;
+  using limiter_comm_tag =
+      SlopeLimiters::Tags::LimiterCommunicationTag<metavariables>;
+
+  const Mesh<2> mesh{3, Spectral::Basis::Legendre,
+                     Spectral::Quadrature::GaussLobatto};
+
+  //      xi      Block       +- xi
```

Comment:
Okay, I know you copied this stick figure from another test, but I think I've finally figured out why I am always confused by it.  If I understand the figure, in Block 0: upper_xi points north/up, and upper_eta points west/left; while in Block 1:  upper_xi points east/right, and upper_eta points south/down.   If so, the logical coordinate system in Block 1 is left-handed and I find this to be confusing, and do not see any reason why this would be desirable.   Not something we need to hold up this PR, but something we should discuss in general...

Reply:
I also interpret the figure in this way. Are we making an effort (in particular, in the more complex 3D domains) to maintain right-handed coordinates systems always? If so I agree, this could be updated.